src/systems/optical_tactile_plugin/OpticalTactilePlugin.cc

```diff
@@ -805,10 +805,12 @@ void OpticalTactilePluginPrivate::ComputeNormalForces(
       // Forces buffer is composed of XYZ coordinates, while _msg buffer is
       // made up of XYZRGB values
       bufferIndex = j * (_msg.row_step() / 2) + i * (_msg.point_step() / 2);
-      normalForcesBuffer.get()[bufferIndex] = normalForce.X();
-      normalForcesBuffer.get()[bufferIndex + sizeof(float)] = normalForce.Y();
+      normalForcesBuffer.get()[bufferIndex] =
```

Comment:
casting a `float` to `char` seems suspect. I think what we want is probably the following:
```suggestion
      std::memcpy(&normalForcesBuffer.get()[bufferIndex], &normalForce.X(), sizeof(float));
```

Here's an example that shows the problem: https://godbolt.org/z/r48bTveYx

Reply:
Agree that the suggestion makes more sense.